components/core/DataView.js

```diff
@@ -239,6 +211,7 @@ export default class DataView extends React.Component {
       window.addEventListener("remote-update-carousel", this._handleUpdate);
     }
 
+    window.addEventListener("scroll", this._handleScroll);
```

Comment:
It looks like it's not actually using the debounce right now. B/c you have an event listener here that calls `handleScroll`, it calls `handleScroll` everytime you scroll. Instead, you want to attach the event listener to `handleCheckScroll`, which cancels the timer every time you scroll (and upon stopping scrolling, then calls handleScroll)

Reply:
Looks like this still isn't fixed. I saw you added an extra await this._handleCheckScroll below it, but that's different from changing this window.addEventListener("scroll", this._handleScroll); to window.addEventListener("scroll", this._handleCheckScroll);

See my slack message for more details